src/renderer.ts

```diff
@@ -152,6 +157,503 @@ export class DefaultCellRenderer implements ICellRendererFactory {
+
+    return (ctx: CanvasRenderingContext2D, d: IDataRow, i: number) => {
+      const data = col.getValue(d.v, d.dataIndex);
+      var scale = verticalBarScale(dataInfo, defaultScale, context.rowHeight(i));
+
+      data.forEach((d, i) => {
+        var xpos = (i * cellDimension);
+        var ypos = verticalBarYpos(dataInfo, d, scale, context.rowHeight(i));
+        ctx.fillStyle = colorScale(d);
+        ctx.fillRect(xpos, ypos, cellDimension, verticalBarHeight(dataInfo, d, scale, context.rowHeight(i)));
+      });
+    };
+  }
+
+}
+
+class BoxplotCellRenderer implements ICellRendererFactory {
+
+  createSVG(col: IBoxPlotColumn & Column, context: IDOMRenderContext): ISVGCellRenderer {
+    const userSort = col.getUserSortBy();
+    const topPadding = Math.max(context.option('rowPadding', 1), 2);
+    const bottomPadding = Math.max(context.option('rowPadding', 1), 2);
+    const dataInfo = col.getDataInfo();
+    const scale = d3.scale.linear().domain([dataInfo.min, dataInfo.max]).range([0, col.getWidth()]);
+    return {
+
+      template: `<g class="boxplotcell"></g>`,
+      update: (n: SVGGElement, d: IDataRow, i: number) => {
+
+        const boxdata = col.getBoxPlotData(d.v, d.dataIndex, scale);
+        if ((boxdata) === null || undefined) {
+          return;
+        }
+        const sortPosition = {
+          min: boxdata.min,
+          max: boxdata.max,
+          median: boxdata.median,
+          q1: boxdata.q1,
+          q3: boxdata.q3
+        };
+
+        const minPos = sortPosition.min, maxPos = sortPosition.max, medianPos = sortPosition.median, q3Pos = sortPosition.q3, q1Pos = sortPosition.q1;
+        const rect = d3.select(n).selectAll('rect').data([col.getValue(d.v, d.dataIndex)]);
+        rect.enter().append('rect');
+        rect
+          .attr('class', 'boxplotrect')
+          .attr({
+            y: topPadding,
+            x: (q1Pos),
+            width: (q3Pos - q1Pos),
+            height: (d, i) => (context.rowHeight(i) - (topPadding + bottomPadding))
+          });
+        rect.exit().remove();
+
+        const pathAll = d3.select(n).selectAll('path.boxplotallpath').data([<any>col.getValue(d.v, d.dataIndex)]);
+        pathAll.enter().append('path');
+        const bottomPos = (context.rowHeight(i) - bottomPadding);
+        const middlePos = (context.rowHeight(i)) / 2;
+        pathAll
+          .attr('class', 'boxplotallpath')
+          .attr('d', () =>
+            'M' + minPos + ',' + middlePos + 'L' + (q1Pos) + ',' + middlePos +
+            'M' + minPos + ',' + topPadding + 'L' + minPos + ',' + (bottomPos) +   //minimum line
+            'M' + medianPos + ',' + topPadding + 'L' + medianPos + ',' + bottomPos +   //median line
+            'M' + (q3Pos) + ',' + middlePos + 'L' + (maxPos) + ',' + middlePos +
+            'M' + maxPos + ',' + topPadding + 'L' + maxPos + ',' + bottomPos   // maximum line
+          );
+
+        d3.select(n).select('.boxplotsortpath').remove();
```

Comment:
this is to redraw the red line in the boxplot when the ascending for consecutive sort criteria is same.